modules/tinymce/src/core/main/ts/api/html/SaxParser.ts

```diff
@@ -629,6 +629,18 @@ function SaxParser(settings?: SaxParserSettings, schema = Schema()): SaxParser {
     }
   };
 
+  /**
+   * Parses the specified HTML string and executes the callbacks for each item it finds.
+   *
+   * @example
+   * SaxParser({...}).parse('<b>text</b>');
+   * @method parse
+   * @param {String} html Html string to sax parse.
+   */
+  const parse = (html: string, format: ParserFormat = 'html') => {
+    parseHtml(extractBase64DataUris(html), format);
```

Comment:
Again, this doesn't look like it's really the right place for this sort of logic. The `SaxParser` is a streaming API, so I believe if we want to extract this, it should be done in `DomParser` instead before we pass it to the `SaxParser`, as that's more relevant to processing it into DOM elements.

Reply:
That is not possible since the SaxParser is what is causing the bug with the backtracking. So we need to process away the huge blobs before we parse or it will fail.

Reply:
That's what I said though? As you'd preprocess it before using SaxParser, as atm it's trying to do that in a streaming API which isn't the right place, since it's concern is to just fire events/callback when encountering certain data.

Reply:
Looking at this, more this probably is okay as it just extracts the data uris and replaces it with tokens.

So it's really the `dataUris` callback that I don't like here, so how about instead we keep this but remove `dataUris` from `SaxParser`. Then we just replace the token with the actual URI when firing the `start` callback that contains the attribute list. From there we can then manipulate the URIs and convert them to blobs from within the `start` callback in DomParser. Thoughts?

Reply:
Right I can rework this so it puts back the base64 url in the attribute callback then we can get rid of the dataUri callback but that would mean we would have to parse the dara url again in the DOMParser mime and data but that should be pretty fast.

Reply:
On mobile didn't see you comment there we are on the same page.

Reply:
Yeah, the performance and having to re-parse the mime type is a valid point. Maybe we should just instead go with making it a setting and calling it `data_uri_resolver` even if that does mix the concerns a bit 🤷‍♀ Your call on what you think is best 👍